src/dune/preprocessing.ml

```diff
@@ -566,20 +566,36 @@ type t = (Module.t -> lint:bool -> Module.t) Per_module.t
 
 let dummy = Per_module.for_all (fun m ~lint:_ -> m)
 
+let is_preprocessing t =
+  Per_module.fold t ~init:true ~f:(fun p acc ->
+      match p with
+      | Preprocess.Without_future_syntax.No_preprocessing -> false
+      | _ -> acc)
+
 let make sctx ~dir ~expander ~dep_kind ~lint ~preprocess ~preprocessor_deps
     ~lib_name ~scope =
+  let preprocess =
+    Per_module.map preprocess ~f:(fun pp ->
+        Dune_file.Preprocess.remove_future_syntax ~for_:Compiler pp
+          (Super_context.context sctx).version)
+  in
   let preprocessor_deps =
-    Build.memoize "preprocessor deps" preprocessor_deps
+    let loc, deps = preprocessor_deps in
+    ( if (not (is_preprocessing preprocess)) && not (List.is_empty deps) then
```

Comment:
parentheses around the first `not` are not needed.

Reply:
Fixed.